xquant/common/framework_report_utils.py

```diff
@@ -0,0 +1,259 @@
+            dataset (Callable): Dataset used for evaluation.
+            custom_metrics_intermediate (Dict[str, Callable], optional): Custom metrics for intermediate layers. Defaults to None.
+            is_validation (bool, optional): Flag indicating if this is a validation dataset. Defaults to False.
+
+        Returns:
+            Dict[str, Dict[str, float]]: A dictionary of computed metrics for intermediate layers.
+        """
+        Logger.critical(f"get_metric_on_intermediate is not implemented.")
+
+    def get_default_metrics(self) -> Dict[str, Callable]:
+        """
+        Retrieve the default metrics for evaluation.
+
+        Returns:
+            Dict[str, Callable]: A dictionary of default metric functions.
+        """
+        Logger.critical(f"get_default_metrics is not implemented.")
+
+    def create_float_folded_model(self,
+                                  float_model: Any,
+                                  representative_dataset: Any):
+        """
+        Create a folded version of the floating-point model.
+
+        Args:
+            float_model (Any): The floating-point model.
+            representative_dataset (Any): Representative dataset used during folding.
+
+        Returns:
+            Any: The folded floating-point model.
+        """
+        Logger.critical(f"create_float_folded_model is not implemented.")
+
+    def wrapped_dataset(self,
+                        dataset: Any,
+                        is_validation: bool,
+                        device: str):
+        """
+        Wrap the dataset for handling labeled and unlabeled datasets.
+
+        Args:
+            dataset (Any): The dataset to wrap.
+            is_validation (bool): Flag indicating if this is a validation dataset.
+            device (str): Device to use for evaluation.
+
+        Returns:
+            Any: Wrapped dataset ready for evaluation.
+        """
+        Logger.critical(f"wrapped_dataset is not implemented.")
+
+    def get_float_to_quantized_compare_points(self,
+                                              quantized_model: Any,
+                                              float_model: Any) -> Dict[str, str]:
+        """
+        Get comparison points between the floating-point and quantized models.
+
+        Args:
+            quantized_model (Any): The quantized model.
+            float_model (Any): The floating-point model.
+
+        Returns:
+            Dict[str, str]: A dictionary mapping comparison points between the two models.
+        """
+        Logger.critical(f"get_float_to_quantized_compare_points is not implemented.")
+
+    def compute_metrics(self,
+                        predictions: Tuple[Any, Any],
+                        custom_metrics_output: Dict[str, Callable] = None) -> Dict[str, float]:
+        """
+        Compute metrics based on predictions.
+
+        Args:
+            predictions (Tuple[Any, Any]): A tuple of predictions from the floating-point and quantized models.
```

Comment:
A tuple of predictions of the two compared models /  A tuple of predictions to compute the difference metrics on.

(maybe in the future it won't necessarily be fp model and quant model so no need to restrict to that)